zproject/backends.py

```diff
@@ -1966,6 +1992,24 @@ def auth_complete(self, *args: Any, **kwargs: Any) -> Optional[HttpResponse]:
             return None
 
 
+class ZulipSAMLIdentityProvider(SAMLIdentityProvider):
+    def get_user_details(self, attributes: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Overriden to support plumbing of additional Attributes
+        from the SAMLResponse.
+        """
+        result = super().get_user_details(attributes)
+
+        extra_attr_names = self.conf.get("extra_attrs", [])
+        result["extra_attrs"] = {}
+        for extra_attr_name in extra_attr_names:
+            result["extra_attrs"][extra_attr_name] = self.get_attr(
+                attributes, None, extra_attr_name
+            )
```

Comment:
Can we pass these as named kwargs or something to self-document the meaning of the parameters like `None` here?

Reply:
Yup, done
```
            result["extra_attrs"][extra_attr_name] = self.get_attr(
                attributes=attributes, conf_key=None, default_attribute=extra_attr_name
            )
```